src/sentiment.py

```diff
@@ -0,0 +1,302 @@
+    mask_lens.columns = ["length"]
+    data = data.loc[mask_lens["length"] < 200]
+    data = data.drop(columns=["CourseId", "Label"])
+
+    data = data.dropna(axis=0, how="any")  # tokenizer requires no None reviews
+
+    print(data.describe())
+    data_reserve = data.sample(frac=examples_set_aside, axis=0)
+    data = data.loc[~(data["Review"].isin(data_reserve["Review"]))]
+    # Split into 70/15/15
+    train_text, temp_text, train_labels, temp_labels = train_test_split(
+        data["Review"],
+        data["Class"],
+        random_state=2021,
+        test_size=0.3,
+        stratify=data["Class"],
+    )
+
+    val_text, test_text, val_labels, test_labels = train_test_split(
+        temp_text, temp_labels, random_state=2021, test_size=0.5, stratify=temp_labels
+    )
+
+    # import BERT-base pretrained model
+    bert = AutoModel.from_pretrained("bert-base-uncased")  # This works
+
+    # Load the BERT tokenizer
+    tokenizer = BertTokenizerFast.from_pretrained("bert-base-uncased")  # This works
+
+    tokens_train = tokenizer.batch_encode_plus(
+        train_text.tolist(), max_length=pad_len, pad_to_max_length=True, truncation=True
+    )
+    tokens_val = tokenizer.batch_encode_plus(
+        val_text.tolist(), max_length=pad_len, pad_to_max_length=True, truncation=True
+    )
+    tokens_test = tokenizer.batch_encode_plus(
+        test_text.tolist(), max_length=pad_len, pad_to_max_length=True, truncation=True
+    )
+    train_seq = torch.tensor(tokens_train["input_ids"])
+    train_mask = torch.tensor(tokens_train["attention_mask"])
+    train_y = torch.tensor(train_labels.tolist())
+    val_seq = torch.tensor(tokens_val["input_ids"])
+    val_mask = torch.tensor(tokens_val["attention_mask"])
+    val_y = torch.tensor(val_labels.tolist())
+    test_seq = torch.tensor(tokens_test["input_ids"])
+    test_mask = torch.tensor(tokens_test["attention_mask"])
+    test_y = torch.tensor(test_labels.tolist())
+
+    # define a batch size
+    batch_size = 64
+
+    # wrap tensors
+    train_data = TensorDataset(train_seq, train_mask, train_y)
+    # sampler for sampling the data during training
+    train_sampler = RandomSampler(train_data)
+    # dataLoader for train set
+    train_dataloader = DataLoader(
+        train_data, sampler=train_sampler, batch_size=batch_size
+    )
+    # wrap tensors
+    val_data = TensorDataset(val_seq, val_mask, val_y)
+    # sampler for sampling the data during training
+    val_sampler = SequentialSampler(val_data)
+    # dataLoader for validation set
+    val_dataloader = DataLoader(val_data, sampler=val_sampler, batch_size=batch_size)
+
+    # freeze the BERT architecture
+    for param in bert.parameters():
+        param.requires_grad = False
+
+    # pass the pre-trained BERT to our define architecture
+    model = BERT_architecture(bert)
+
+    # push the model to GPU - I THINK DONT DO THIS
+    model = model.to(device)
```

Comment:
I think this is right

Reply:
Yeah it was just old comment i didn't delete